### PR TITLE
Add Bitbucket Metadata for Release

### PR DESCRIPTION
**Pull request by InTheCloudDan**
Add Bitbucket metadata and scripts to automate release. These are not wired into the release process yet.

The update script will pick up the environment variable `LD_RELEASE_VERSION` but is generally expected to be run manually by a user before the release in which case it can be run with `.ldrelease/build/update-bitbucket.sh 0.0.1`. That can be changed in future PRs.

### CODE REVIEW

File: build/metadata/bitbucket/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+| LD_IGNORE_SERVICE_ERRORS | If enabled, the scanner will terminate with exit code 0 when the LaunchDarkly API is unreachable or returns an unexpected response. Default: `false` |
+| LD_LOOKBACK | Sets the number of Git commits to search in history for whether a feature flag was removed from code. May be set to 0 to disabled this feature. Setting this option to a high value will increase search time. Defaults to 10 |
+
+## Details
+LaunchDarkly's Code references feature allows you to find source code references to your feature flags within LaunchDarkly. This makes it easy to determine which projects reference your feature flags, and makes cleanup and removal of technical debt easy. For more information, visit our [documentation](https://docs.launchdarkly.com/v2.0/docs/git-code-references). For documentation on the source code of this pipe, see the [source repo](https://github.com/launchdarkly/ld-find-code-refs).
+
+
+## Prerequisites
+A LaunchDarkly personal access token with writer-level access, or access to the `code-reference-repository` [custom role](https://docs.launchdarkly.com/v2.0/docs/custom-roles) resource.
```
Comment (jazanne) [inline build/metadata/bitbucket/README.md:44]:
```suggestion
A LaunchDarkly personal access token with writer-level access, or access to the `code-reference-repository` [custom role](https://docs.launchdarkly.com/home/members/custom-roles) resource.
```

File: build/metadata/bitbucket/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+script:
+  - pipe: - pipe: launchdarkly/ld-find-code-refs-pipe:2.4.0
+    environment:
+      LD_ACCESS_TOKEN: $LD_ACCESS_TOKEN
+      LD_PROJ_KEY: $LD_PROJ_KEY
+      LD_CONTEXT_LINES: "3"
+```
+
+## Support
+If youâ€™d like help with this pipe, or you have an issue or feature request, [submit a request](https://support.launchdarkly.com/hc/en-us/requests/new).
```
Comment (jazanne) [inline build/metadata/bitbucket/README.md:67]:
```suggestion
If you'd like help with this pipe, or you have an issue or feature request, [submit a request](https://support.launchdarkly.com/hc/en-us/requests/new).
```

File: build/metadata/bitbucket/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+If youâ€™re reporting an issue, please include:
```
Comment (jazanne) [inline build/metadata/bitbucket/README.md:69]:
```suggestion
If you're reporting an issue, please include:
```

File: build/metadata/bitbucket/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+| LD_DEBUG | Enables verbose debug logging. Default: `false`|
+| LD_DEFAULT_BRANCH | The git default branch. The LaunchDarkly UI will default to display code references for this branch. Default: `master`. |
+| LD_IGNORE_SERVICE_ERRORS | If enabled, the scanner will terminate with exit code 0 when the LaunchDarkly API is unreachable or returns an unexpected response. Default: `false` |
+| LD_LOOKBACK | Sets the number of Git commits to search in history for whether a feature flag was removed from code. May be set to 0 to disabled this feature. Setting this option to a high value will increase search time. Defaults to 10 |
+
+## Details
+LaunchDarkly's Code references feature allows you to find source code references to your feature flags within LaunchDarkly. This makes it easy to determine which projects reference your feature flags, and makes cleanup and removal of technical debt easy. For more information, visit our [documentation](https://docs.launchdarkly.com/v2.0/docs/git-code-references). For documentation on the source code of this pipe, see the [source repo](https://github.com/launchdarkly/ld-find-code-refs).
```
Comment (jazanne) [inline build/metadata/bitbucket/README.md:40]:
```suggestion
LaunchDarkly's Code references feature allows you to find source code references to your feature flags within LaunchDarkly. This makes it easy to determine which projects reference your feature flags, and makes cleanup and removal of technical debt easy. For more information, visit our [documentation](https://docs.launchdarkly.com/home/code/code-references). For documentation on the source code of this pipe, see the [source repo](https://github.com/launchdarkly/ld-find-code-refs).
```

File: .ldrelease/update-bitbucket.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (yusinto) [inline .ldrelease/update-bitbucket.sh:1]: Nit: Perhaps renaming this to update-bitbucket-metadata.sh can make the intent of this file clearer?

File: .ldrelease/update-version.sh [hunk 6]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 VERSION_GO=internal/version/version.go
 VERSION_GO_TEMP=${VERSION_GO}.tmp
 sed "s/const Version =.*/const Version = \"${LD_RELEASE_VERSION}\"/g" ${VERSION_GO} > ${VERSION_GO_TEMP}
```
Comment (yusinto) [inline .ldrelease/update-version.sh:7]: Should we add a comment here to remind us to invoke `update-bitbucket-metadata.sh`?

```sh
# TODO: update bitbucket metadata when techops approves the ci bot
# https://launchdarkly.atlassian.net/servicedesk/customer/portal/11/TO-5131
# ./update-bucket-metadata.sh
```

**Review by yusinto — APPROVED** [review]

**Pull request merged**